packages/monitor-v2/src/llm-bot/common.ts

```diff
@@ -187,3 +198,82 @@ export abstract class LLMStrategy<I extends OptimisticOracleRequest, R extends O
     return this.results;
   }
 }
+
+/**
+ * Enum for options available for handling strategy results.
+ */
+export enum StrategyHandlerOptions {
+  Dispute = "Dispute",
+  Propose = "Propose",
+  Log = "Log",
+  Backtest = "Backtest",
+}
+
+/**
+ * Class for handling the results of an LLMStrategy.
+ * This can be used to execute further actions based on the strategy results, such as disputing or proposing new requests, logging, or backtesting.
+ * @template R The type of the results from the LLMStrategy.
+ */
+export abstract class OptimisticOracleStrategyHandler<R extends OptimisticOracleRequest> {
+  protected strategyResults: R[];
+
+  /**
+   * Creates a new OptimisticOracleStrategyHandler.
+   * @param strategyResults The results from an LLMStrategy to be handled.
+   */
+  constructor(strategyResults: R[]) {
+    this.strategyResults = strategyResults;
+  }
+
+  /**
+   * Processes the results of the strategy based on the provided option.
+   * This is an abstract method that should be overridden by subclasses, defining the specific processing logic.
+   * @param option One of the options in the StrategyHandlerOptions enum, which defines the action to take on the results.
+   * @returns A Promise that resolves once the processing is complete.
+   */
+  abstract process(option: StrategyHandlerOptions): Promise<void>;
+
+  /**
+   * Initiates a dispute for a specific strategy result.
+   * This method should be overridden by implementations that need to dispute requests.
+   * @param result The specific result to dispute.
+   * @returns A Promise that resolves once the dispute has been initiated.
+   */
+  protected async disputeRequest(result: R): Promise<void> {
```

Comment:
Do we need these methods/implementations in the interface/base class? Would this structure be important for the caller to know about/be aware of?